parley/src/layout/editor.rs

```diff
@@ -15,12 +15,27 @@ use alloc::{sync::Arc, vec::Vec};
 
 #[derive(Copy, Clone, Debug)]
 pub enum ActiveText<'a> {
-    /// The selection is empty and the cursor is a caret; this is the text of the cluster it is on
+    /// The selection is empty and the cursor is a caret; this is the text of the cluster it is on.
     FocusedCluster(Affinity, &'a str),
-    /// The selection contains this text
+    /// The selection contains this text.
     Selection(&'a str),
 }
 
+/// Opaque representation of a generation.
+// Overflow handling: the generations are only compared,
+// so wrapping is usually fine, except that in the unlikely
+// event that exactly `u32::MAX` generations happen before the
+// application gets around to drawing `Generation(0)`.
```

Comment:
```suggestion
// so wrapping is fine. This could only fail if exactly
// `u32::MAX` generations happen between drawing
// operations. This is implausible and so can be ignored.
```